### ISSUE TITLE
Add dedicated course link (with modal)

### ISSUE THREAD

**Issue by eamonma** [sqrl-planner/sqrl-client#26]
Add the ability to share a course as a link, such that when someone opens this link a modal displays information regarding this course. The motivation for this feature is to enable easier coordination with friends, i.e. "take this course with me".

- Potentially add ability to share link containing section(s) selected

**Comment by galacticglum** [comment]
How is this any different from just sending the course code to friend? Not sure what benefit this would provide. I like the idea of packaging selected sections into the link. I think that makes it a somewhat worthwhile feature.

**Comment by eamonma** [comment]
I agree, I think inherently sharing selected sections makes this much more valuable.

**Comment by galacticglum** [comment]
Possible url spec for implementation could be:

```
https://app.sqrlplanner.com/course/share?id=<:id>&sections=<:id1>,<:id2>,...
```
which will open a page including the course information with a button that allows them to add it to their timetable.

**Comment by eamonma** [comment]
The latter looks good to me. I tend to avoid query params if possible—for no reason other than it looks ugly. Also, I'd probably extend that to specify lecture section, tutorial section, and practical section, particularly because I don't believe we have unique ids for sections in case of duplicate section ids, such as CSC110.

<img width="443" alt="image" src="https://user-images.githubusercontent.com/16643012/171900811-5ebec112-7ec8-499d-a238-d806fdc1483a.png">

**Comment by galacticglum** [comment]
I thought the db had unique ids for them? Sure, we can avoid query params, but I'm not sure how you would do it without.

**Comment by eamonma** [comment]
I meant for the course id. So in this case, we would have
`https://app.sqrlplanner.com/course/<:id>/share?lecture=<:lecid>&tutorial=<:tutid>,...`

**Comment by galacticglum** [comment]
I think I'd prefer to not have query params for each section type.

**Comment by galacticglum** [comment]
BLOCKED by [issue #35 in gator](https://github.com/sqrl-planner/gator/issues/35).

**Comment by galacticglum** [comment]
UPDATE: Issue is unblocked.
